tools/torch_quant/torch_quant/quantizer.py

```diff
@@ -342,3 +357,44 @@ def quantize(self, model: nn.Module) -> nn.Module:
         for name, traced in trace_mapping.items():
             self.quantize_gm(name, traced.gm, traced.m)
         return copy_and_replace(model, trace_mapping)
+
+    def export_mnn_params_gm(
+        self, name: str, gm: GraphModule, root: nn.Module, ob_types: ObserverTypes
+    ) -> None:
+        mf = submodule_filter(self.module_filter, name) if self.module_filter else None
+        ctx = GraphModContext(
+            gm=gm,
+            root=root,
+            module_filter=mf,
+            act_ob_ctr=ob_types.act_ob_ctr,
+            w_ob_ctr=ob_types.w_ob_ctr,
+            bias_ob_ctr=ob_types.bias_ob_ctr,
+        )
+        ctx.modify_graph([
+            set_qconfig,
+            fuse_modules,
+            insert_act_observer,
+            insert_w_observer,
+            quantizable_module_to_observed,
+        ])
+
+    def export_mnn_params(
+        self, model: nn.Module, dummy_input: Union[Tuple[Any, ...], torch.Tensor]
+    ) -> Pipeline:
+        settings = ['qscheme', 'dtype']
+        kwds = lambda keywords: {k: v for k, v in keywords.items() if k in settings}
+        act_ob_ctr = partial(FakeQuantizer, **kwds(self.act_ob_ctr.keywords))
+        w_ob_ctr = partial(FakeQuantizer, **kwds(self.w_ob_ctr.keywords))
```

Comment:
why need another observer type (FakeQuantizer) just for export? most observer type (e.g. HistogramObserver ) already implement fake quant forward behavior?

Reply:
Done. MNN quantization needs to export onnx model, while there's no need for other backend. I thought it's straightforward by using a new observer type (FakeQuantizer). However, it may be better to implement it in base class (Observer)?